app/models/topic_tracking_state.rb

```diff
@@ -217,14 +245,27 @@ def self.publish_read(topic_id, last_read_post_number, user_id, notification_lev
 
   def self.publish_dismiss_new(user_id, topic_ids: [])
     message = {
-      message_type: "dismiss_new",
+      message_type: DISMISS_NEW_MESSAGE_TYPE,
       payload: {
         topic_ids: topic_ids
       }
     }
     MessageBus.publish(self.unread_channel_key(user_id), message.as_json, user_ids: [user_id])
   end
 
+  def self.new_filter_sql
+    TopicQuery.new_filter(Topic, "xxx").where_clause.ast.to_sql.gsub!("'xxx'", treat_as_new_topic_clause) +
```

Comment:
This is really hacky. Is there no way to do this in a cleaner way? Doing a substitution in SQL seems error prone to me.

Reply:
This was historical. I added a different parameter to `TopicQuery.new_filter` for this now instead.